lasagne/layers/conv.py

```diff
@@ -919,26 +950,301 @@ def get_output_shape_for(self, input_shape):
+    convolution, fractionally-strided convolution or deconvolution in the
+    literature) on its input and optionally adds a bias and applies an
+    elementwise nonlinearity.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape. The
+        output of this layer should be a 4D tensor, with shape
+        ``(batch_size, num_input_channels, input_rows, input_columns)``.
+
+    num_filters : int
+        The number of learnable convolutional filters this layer has.
+
+    filter_size : int
+        An integer specifying the size of the filters.
+
+    stride : int
+        An integer specifying the stride of the
+        transposed convolution operation. For the transposed convolution, this
+        gives the dilation factor for the input -- increasing it increases the
+        output size.
+
+    crop : int, iterable of int, 'full', 'same' or 'valid' (default: 0)
+        By default, the transposed convolution is computed where the input and
+        the filter overlap by at least one position (a full convolution). When
+        ``stride=1``, this yields an output that is larger than the input by
+        ``filter_size - 1``. It can be thought of as a valid convolution padded
+        with zeros. The `crop` argument allows you to decrease the amount of
+        this zero-padding, reducing the output size. It is the counterpart to
+        the `pad` argument in a non-transposed convolution.
+
+        A single integer results in symmetric cropping of the given size on all
+        borders, a tuple of two integers allows different symmetric cropping
+        per dimension.
+
+        ``'full'`` disables zero-padding. It is is equivalent to computing the
+        convolution wherever the input and the filter fully overlap.
+
+        ``'same'`` pads with half the filter size (rounded down) on both sides.
+        When ``stride=1`` this results in an output size equal to the input
+        size. Even filter size is not supported.
+
+        ``'valid'`` is an alias for ``0`` (no cropping / a full convolution).
+
+        Note that ``'full'`` and ``'same'`` can be faster than equivalent
+        integer values due to optimizations by Theano.
+
+    untie_biases : bool (default: False)
+        If ``False``, the layer will have a bias parameter for each channel,
+        which is shared across all positions in this channel. As a result, the
+        `b` attribute will be a vector (1D).
+
+        If True, the layer will have separate bias parameters for each
+        position in each channel. As a result, the `b` attribute will be a
+        3D tensor.
+
+    W : Theano shared variable, expression, numpy array or callable
+        Initial value, expression or initializer for the weights.
+        These should be a 4D tensor with shape
+        ``(num_input_channels, num_filters, filter_rows, filter_columns)``.
```

Comment:
This is correct now. I wonder whether my previous comment (https://github.com/Lasagne/Lasagne/pull/850/files#r123304988) was originally attached to the 1D version and github messed up, or whether I accidentally thought it was about the 1D version.